microsoft-edge/webview2/concepts/user-data-folder.md

```diff
@@ -1,65 +1,637 @@
+
+
+**What type of platform app a ClickOnce app wraps:**
+
+A ClickOnce app usually contains a .NET app.
 
 
 <!-- ====================================================================== -->
-## Best practices
+<!-- default UDF location -->
+
+# [WinUI 2 (UWP)](#tab/uwp)
+
+<!--
+**What is the default UDF location?**
+-->
+On this platform, the default UDF location is the `ApplicationData\LocalFolder` subfolder in the package's folder.
+
+
+**Should you use default or custom UDF location?**
+
+Use the default UDF location.
+
+
+**Why the default location is writeable:**
+
+WinUI 2 (UWP) is a packaged platform; it runs in the sandbox and is packaged to run in the sandbox, not usually as separate files, but as an app bundle.
 
-User data folders are created automatically by WebView2.  WebView2 developers control the lifetime of the user data folder.  If your application re-uses user data from application sessions, consider saving the user data folders, otherwise you can delete them.  Consider the following scenarios when deciding how to manage your user data folders:
+A WinUI 2 (UWP) app is per-user, and has Write access under the installed location.
 
-*  If the same user uses your application repeatedly, and the web content of the application relies on the user's data, save the user data folder.
+For UWP, the `.exe` is in a protected location that the installer can write to.
 
-*  If multiple users use your application repeatedly, create a new user data folder for each new user, and save the user data folder of each user.
+At run time, during the session only, your WebView2 host app is given access to the local folder, so the WebView2 control is able to create the UDF there and write to it.
 
-*  If your application doesn't have repeat users, create a new user data folder for each user, and delete the previous user data folder.
+WebView2 checks for that runtime, and creates the UDF in that writeable location.
 
-*  Instead of deleting the user data folder, clear browsing data from the user data folder.  For example, clear user data and history when a user signs out.  See [clear browsing data from the user data folder](clear-browsing-data.md).
+
+**Packaging:**
+
+A WinUI 2 (UWP) app is self-contained regarding dependencies and for deployment (DLLs), although it's not packaged into a single file.
+
+A WinUI 2 (UWP) app is per-user, and has Write access under the installed location.
 
 
 <!-- ====================================================================== -->
-## Create user data folders
+<!-- default UDF location -->
+
+# [WinUI 3](#tab/winui3)
+
+<!--
+**What is the default UDF location?**
+-->
+On this platform, the default UDF location is the `ApplicationData\LocalFolder` subfolder in the package's folder.
+
+
+**Should you use default or custom UDF location?**
+
+In most cases, you should use the default UDF location.
+
+OR:<!--todo: Dev to investigate which is the case -->
```

Comment:
@DavidShoe - On WinUI 3, do we advising using the default UDF location, or custom UDF location?

Reply:
@DavidShoe - please sync / check across the two WinUI 3 sections.  There are 3 h2 sections that contain tabs, and the first two of these h2 sections - in their WinUI 3 tabs - both need to contain a statement saying whether to use default or custom UDF location.  I'll look at it in VS Code, and check w/ you to make sure the right statement is in both sections, sync'd.  It's possible you told me about one section, and I need to propagate your answer into the other section.

Reply:
WinUI 2 & 3 use the same recomendations and restrictions at this time.  This may change in the future as WinUI 3 matures so keeping them separate makes sense.